src/lib/autoLauncher.ts

```diff
@@ -0,0 +1,141 @@
+import * as app from "@tauri-apps/api/app"
+import * as os from "@tauri-apps/api/os"
+import { invoke } from '@tauri-apps/api/tauri'
+import { AutoLaunchParams, ChildReturnData } from './types'
+import * as fs from "@tauri-apps/api/fs"
+import * as native from './native'
+
+
+type osAL = typeof macAL | typeof winAL | typeof linAL | typeof nullAL
+
+const nullAL = {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async enable({ appName, appPath, hidden }: AutoLaunchParams): Promise<ChildReturnData> {
+    return { stdout: [], stderr: [] }
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async disable(appName: string): Promise<ChildReturnData> {
+    return { stdout: [], stderr: [] }
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async isEnabled(appName: string): Promise<boolean> {
+    return false
+  }
+}
+
+const linAL = {
+
+  async enable({ appName, appPath, hidden }: AutoLaunchParams): Promise<ChildReturnData> {
+    const response: ChildReturnData = { stderr: [], stdout: [] }
+    const hiddenArg = hidden ? ' --hidden' : '';
+
+    const contents = `
+    [Desktop Entry]
+    Type=Application
+    Version=1.0
+    Name=${appName}
+    Comment=${appName}startup script
+    Exec=${appPath}${hiddenArg}
+    StartupNotify=false
+    Terminal=false
```

Comment:
```suggestion
[Desktop Entry]
Type=Application
Name=${appName}
Comment=${appName} startup script
Exec=${appPath}${hiddenArg}
Icon=PATH_TO_APP_ICON
```

Reply:
The changes implemented are not the changes I suggested. Please don't resolve things unless they are absolutely 100% resolved, it is annoying to go through resolved threads and unresolve them over and over again.

Reply:
ah my mistake

Reply:
It is more about contents than spacing actually

Reply:
ok updated